File: XEPSO1.py
"""
---------------luyencode.net----------------------------------
---------------Problem: XEPSO1-------------------------------
---------------Author: HeHe---------------------------------
---------------Time: 2024-10-07 20:56:20.136074+07:00---------
"""
in1 = lambda: int(input())
in2 = lambda: list(map(int, input().split()))
MOD = 1000000007
# --------------- Solve ---------------  # 
arr1 = [0, 0, 1, 7, 4, 2, 6, 8]
arr2 = [0, 0, 1, 7, 4, 2, 0, 8]
arr = [6, 2, 5, 5, 4, 5, 6, 3, 7, 6]
# 2 -> 1
# 3 -> 7
# 4 -> 4
# 5 -> 2, 3, 5
# 6 -> 0, 6, 9
# 7 -> 8

def find_max(n):
    if n % 2 == 0:
        return int('1' * (n // 2))
    else:
        return int('7' + '1' * ((n - 3) // 2))

def find_min(val):
    def backtrack(n, count):
        if n == 0:
            return count
        for i in range(min(n, 7), 1, -1):
            c = backtrack(n - i, count + 1)
            if c:
                return c
        return 0
    
    number_of_digits = backtrack(val, 0)
    ans = [float('inf')]
    def backtrack2(n, cur, l, ans):
        if n == 0:
            if l == number_of_digits:
                ans[0] = min(ans[0], cur)
            else:
                return 
        if l == number_of_digits:
            return
        
        if cur == 0:
            for i in range(1, 10):
                backtrack2(n - arr[i], cur * 10 + i, l + 1, ans)
        else:
            for i in range(0, 10):
                backtrack2(n - arr[i], cur * 10 + i, l + 1, ans)
        return

    backtrack2(val, 0, 0, ans)
    return ans[0]
    

max_arr = [0] * 40
min_arr = [0] * 40
# for val in range(2, 40):
#     max_val = find_max(val)
#     min_val = find_min(val)
#     max_arr[val] = max_val
#     min_arr[val] = min_val

t = in1()
for i in range(t):
    n = in1()
    print(find_min(n), find_max(n))

    
##########################################

File: XEPSO1.cpp
/*
---------------luyencode.net----------------------------------
---------------Problem: XEPSO1-------------------------------
---------------Author: HeHe---------------------------------
---------------Time: 2024-10-07 21:15:47.875525+07:00---------
*/
#include <iostream>
#include <iomanip>
#include <vector>
#include <queue>
#include <map>
#include <unordered_map>
#include <set>
#include <unordered_set>
#include <assert.h>
#include <algorithm>
#include <functional>
#include <numeric>
#include <math.h>
#include <cstdint>

#define FAST_IO ios::sync_with_stdio(false);cin.tie(0);cout.tie(0);
using namespace std;

const int MOD = 1000000007;

int main() {
#ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
#endif
    FAST_IO;
    // --------------- Solve -------------------
    vector<int> arr = {6, 2, 5, 5, 4, 5, 6, 3, 7, 6};
    vector<int64_t> min_arr[40], max_arr[40];
    for (int i = 2; i <= 39; ++i) {
        int64_t min_val = INT64_MAX, max_val = INT64_MIN;

        function<void(int, int64_t)> backtrack = [&](int n, int64_t cur) -> void {
            if (n == 0) {
                min_val = min(min_val, cur);
                max_val = max(max_val, cur);
            }
            else {
                if (cur == 0){
                    for (int i = 1; i < 10; ++i) {
                        if (arr[i] <= n) {
                            backtrack(n - arr[i], cur * 10 + i);
                        }
                    }
                }
                else {
                    for (int i = 0; i < 10; ++i) {
                        if (arr[i] <= n) {
                            backtrack(n - arr[i], cur * 10 + i);
                        }
                    }   
                }
            }
        };
        backtrack(i, 0);
        min_arr[i] = min_val;
        max_arr[i] = max_val;
    }

    int t; cin >> t;
    while (t--) {
        int n; cin >> n;
        cout << min_arr[n] << " " << max_arr[n] << '\n';
    }

    ////////////////////////////////////////////
    return 0;
}
